File: backend/internal/model/daily_fortune.go
package model

import "time"

// DailyFortune 每日运势模型
type DailyFortune struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	UserID    string    `json:"user_id" gorm:"index"`
	Date      time.Time `json:"date" gorm:"index"`

	// 基础运势
	Score   int    `json:"score"`   // 综合运势分数 (0-100)
	Summary string `json:"summary"` // 运势概述

	// 详细建议
	Wealth string `json:"wealth"` // 财运
	Career string `json:"career"` // 事业
	Love   string `json:"love"`   // 感情
	Health string `json:"health"` // 健康

	// 幸运指南
	LuckyColor     string `json:"lucky_color"`     // 幸运色
	LuckyNumber    string `json:"lucky_number"`    // 幸运数字
	LuckyDirection string `json:"lucky_direction"` // 幸运方位
	LuckyTime      string `json:"lucky_time"`      // 幸运时辰

	// 宜忌
	Yi []string `json:"yi" gorm:"serializer:json"` // 宜
	Ji []string `json:"ji" gorm:"serializer:json"` // 忌

	// 节气/节日上下文 (生成时的背景)
	SolarTerm string `json:"solar_term"`
	Festival  string `json:"festival"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// DailyFortuneRequest 请求生成每日运势
type DailyFortuneRequest struct {
	UserID string `json:"user_id" binding:"required"`
	Date   string `json:"date"` // YYYY-MM-DD, optional, default to today
}

File: backend-python/app/models/daily_fortune.py
"""每日运势模型"""

from sqlalchemy import Column, Integer, String, Date, Text, JSON
from sqlalchemy.sql import func
from datetime import datetime
from app.core.database import Base


class DailyFortune(Base):
    """每日运势表"""
    __tablename__ = "daily_fortunes"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, nullable=False, index=True)
    fortune_date = Column(Date, nullable=False, index=True)
    
    # 基础运势评分
    overall_score = Column(Integer, nullable=False, default=70)  # 0-100
    wealth_score = Column(Integer, nullable=False, default=70)
    career_score = Column(Integer, nullable=False, default=70)
    love_score = Column(Integer, nullable=False, default=70)
    health_score = Column(Integer, nullable=False, default=70)
    
    # 运势内容
    content = Column(Text, nullable=False, default='')
    
    # 幸运指南
    lucky_color = Column(String(50), nullable=False, default='白色')
    lucky_number = Column(Integer, nullable=False, default=8)
    lucky_direction = Column(String(50), nullable=False, default='东')
    lucky_time = Column(String(50), nullable=False, default='辰时(07:00-09:00)')
    
    # 宜忌（存储为逗号分隔的字符串）
    yi = Column(String(200), nullable=False, default='')
    ji = Column(String(200), nullable=False, default='')
    
    # 节气/节日
    solar_term = Column(String(50), default='')
    festival = Column(String(100), default='')
    
    created_at = Column(Date, server_default=func.current_date(), nullable=False)
    
    def __repr__(self):
        return f"<DailyFortune(user_id={self.user_id}, date={self.fortune_date}, score={self.overall_score})>"
